db/snapshot/backup_task.hh

```diff
@@ -38,17 +39,25 @@ class backup_task_impl : public tasks::task_manager::task::impl {
 
     std::exception_ptr _ex;
     gate _uploads;
-    using comps_vector = utils::small_vector<std::string, sstables::num_component_types>;
+    struct comp_desc {
+        sstring name;
+        uint64_t size;
+    };
+    using comps_vector = utils::small_vector<comp_desc, sstables::num_component_types>;
     using comps_map = std::unordered_map<sstables::generation_type, comps_vector>;
     comps_map _sstable_comps;
     size_t _num_sstable_comps = 0;
     comps_vector _non_sstable_files;
+    // FIXME: this should be configurable, or derived dynamically from
+    // the concurrency level of S3 uploads
+    constexpr static size_t max_concurrency = 32 << 10;
```

Comment:
Do I read it correctly? At most 32k can be uploaded at once?

Reply:
Ouch, you're right, it was supposed to be 32M not K.
And I'll get rid of it in the next version, that will use the sstable dir semaphore.

Reply:
dropped in v3 ([7ab3391](https://github.com/scylladb/scylladb/commit/7ab3391f15e68adfff6187385be438b96bb3d57e))